src/core/types/keys/key_agreement/public_key.rs

```diff
@@ -0,0 +1,243 @@
+use crate::prelude::*;
+use crate::{prelude::*, UniffiCustomTypeConverter};
+use crypto::keys::x25519::PublicKey as X25519PublicKey;
+
+/// A public key for the X25519 key exchange algorithm.
+/// An Ed25519 public key used to verify cryptographic signatures (EdDSA signatures).
+#[serde_as]
```

Comment:
`An Ed25519 public key used to verify cryptographic signatures (EdDSA signatures).` is copy paste error I think.

The doc should probably just be: 
```rust
/// PublicKey on Curve25519 used for key agreement (ECDH) with some `KeyAgreementPrivateKey`.
```